### PR TITLE
use-React-toolkit

### PR DESCRIPTION
**Pull request by muneeb3123**
this Pr includes: 

- Add Redux Toolkit
- Create a directory that will contain all Redux logic (/src/redux)
- Configure a Redux store (/src/redux/store.js)
- Define a slice of state for books that:
Includes an array of books (initial state: empty array)
Includes a reducer that adds a book
Includes a reducer that removes a book

- Define a slice of state for categories that:
Includes an array of categories (initial state: empty array)
Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
+
+const bookSlice = createSlice({
+  name: 'Book',
+  initialState,
+  reducers: {
+    addBook: (state, { payload }) => {
+      state.push(payload);
```
Comment (GzimAsani) [inline src/redux/books/booksSlice.js]: - [ ] This should accessed as  state.book example:

```
    removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },

```

File: src/redux/categories/categories.slice.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+const initialState = [];
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    getStatus: () => 'Under construction',
+  },
+});
```
Comment (GzimAsani) [inline src/redux/categories/categories.slice.js]: - [ ] You   should update the state's categories property based on the value returned by the reducer. Example:

```
  reducers: {
    getStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },

```

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @muneeb3123                :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with creating slices:+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (muneeb3123) [reply src/redux/books/booksSlice.js]: @GzimAsani  kindly connect me on zoom if you want changes.

File: src/redux/categories/categories.slice.js (hunk 2 above)
Comment (muneeb3123) [reply src/redux/categories/categories.slice.js]: If you require any changes while reviewing, I would be grateful if you could connect with me on Zoom. This way, we can discuss the necessary modifications directly, saving time and avoiding any unnecessary delays. Your help in improving my work is greatly appreciated. Thank you.

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @muneeb3123 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/redux/categories/categories.slice.js (hunk 2 above)
Comment (GzimAsani) [reply src/redux/categories/categories.slice.js]: Hi @muneeb3123 sorry for the late reply but I was off last night after this review, I am glad you have figured it out and your project got approved :partying_face:

**Pull request merged**